### ISSUE TITLE
Add capability to set access tokens for users.

### ISSUE THREAD

**Issue by isaacegglestone** [suzuki-shunsuke/go-graylog#244]
Please add ability to:

## API Endpoints

*Generate token for a user:*
/api/api-browser/global/index.html#!/Users/generateNewToken_post_10

*data source for token for a user:*
/api/api-browser/global/index.html#!/Users/listTokens_get_9

*Revoke token for a user via terraform destroy*
/api/api-browser/global/index.html#!/Users/revokeToken_delete_11

Or something of the above nature.

**Comment by suzuki-shunsuke** [comment]
Thank you for your feedback.
I'll check in this weekend.

**Comment by suzuki-shunsuke** [comment]
I have created a pull request https://github.com/suzuki-shunsuke/go-graylog/pull/245

**Comment by suzuki-shunsuke** [comment]
I'm wondering if we should support user tokens API as a Terraform provider.
If we support user tokens API as a Terraform provider, the token would be written in the Terraform state file, but I wonder if it is desiable.

What do you think? @isaacegglestone 

```hcl
resource "graylog_user_token" "foo_test" {
  user_name = "foo"
  token_name = "test"
}
```

**Comment by suzuki-shunsuke** [comment]
https://docs.graylog.org/en/3.2/pages/configuration/rest_api.html#creating-and-using-access-token

>  Also note that users, even administrators, may only manage their own tokens.

Oh, it is difficult to manage tokens with Terraform.

**Comment by suzuki-shunsuke** [comment]
I have released [v11.1.0-0](https://github.com/suzuki-shunsuke/go-graylog/releases/tag/v11.1.0-0).
Please check.
If there is no problem, I'll release v1.11.0 .

**Comment by isaacegglestone** [comment]
Again thanks hugely for the fast turn around on this.
Sometimes I wonder if this is your only job lol!

My use case is that I want to create a monitoring user for prometheus blackbox to hit graylog with.
In this case user generated API tokens is not desirable as its a service type user. I would prefer to have terraform get a generated token and then provide it to the CI that deploys the monitoring component.
Generating one manually means I can't have end to end automation.
Currently I use the users password which is not ideal

So Ideally would be nice to have in terraform provider for that reason.
I can see myself needing many other user tokens and users for other purposes with graylog.

Thanks I will try to check tomorrow my time (Australia time)

**Comment by suzuki-shunsuke** [comment]
Thank you for your explanation.
I understand that you want to monitor Graylog with Prometheus using Graylog's API token,
so you want the monitoring component to call the Graylog API with the monitoring user's API token.
Is it right?

How about the following way?

1. Create a Graylog's user for monitoring with Terraform
2. Generate an API token of the monitoring user manually
3. Set the API token to the CI system such as CircleCI as a secret
4. Deploy the monitoring component with the CI system and set the API token to the monitoring component
5. the monitoring component calls the Graylog API

I think we have to do the step 2 and 3 only once so we don't have to automate it.

I'm sorry if I'm misunderstanding what you want.

Thanks.

**Comment by isaacegglestone** [comment]
RE: I understand that you want to monitor Graylog with Prometheus using Graylog's API,

Essentially but its via this plugin  , not the API directly:
https://github.com/graylog-labs/graylog-plugin-metrics-reporter/tree/master/metrics-reporter-prometheus

ALthough the path does sort of hit the normal API path:
http://graylog.example.com:9000/api/plugins/org.graylog.plugins.metrics.prometheus/metrics

It requires a username and password to authenticate.  

Anyway I think you have done enough that is reasonably requested you can mark this closed now.
It appears to work for me, the part what you did anyway.
I will just go on using the user for now and the password as I don't have any manual steps that way.
But others will no doubt find this useful.

**Comment by suzuki-shunsuke** [comment]
Thank you for your explanation.
I didn't know `metrics-reporter-prometheus`.

Maybe the access token can be used instead of password.

https://docs.graylog.org/en/3.2/pages/configuration/rest_api.html

> The received access token can now be used as username in a request to the Graylog REST API using Basic Auth together with the literal password `token`.

```yaml
basic_auth:
      username: <access token>
      password: token
```

Anyway I close this issue.
I hope your problem would be solved.
Thanks.
